terraform-security.md

```diff
@@ -0,0 +1,19 @@
+# Terraform Security
+
+1. Are you storing your Terraform Code in a version control system (Git or similar)?
+1. Are you using one Terraform workspace for each environment of a given infrastructure component?
+1. Are you using Terraform modules to increase reusability of your infrastructure code?
+1. How do you handle sensitive information in your Terraform scripts?
+    1. How are the credentials of a Terraform service account (JSON) secured?
+    1. Where are these credentials stored?
+    1. Is your Terraform integrated with a secret management tool (HashiCorp’s Vault or similar)?
+    1. Is your Terraform state file encrypted at rest?
+1. Are the permissions of a Terraform service account restricted to minimum?
+    1. What is the list of permissions given to the Terraform service account?
+    1. Are there any separate roles and policies attached?
+1. Are you using Terraform Cloud?
```

Comment:
I would also like to see some questions around Terraform State. How is it protected against accidental disclosure? How is it protected against corruption? Terraform State can not only be stored in Terraform Cloud, but also Git or S3, while still fulfilling these requirements.

Reply:
Absolutely , will add these questions as well . Noted with thanks !!

Reply:
These question now added in the checklist